xdsl/parser/core.py

```diff
@@ -634,6 +634,14 @@ def parse_region(self, arguments: Iterable[Argument] | None = None) -> Region:
             self.raise_error("Expected region!")
         return region
 
+    def parse_regions(self) -> list[Region]:
```

Comment:
Can you rename `parse_regions` to `parse_region_list`? To make it consistent with MLIR.
Also, can you add documentation like above on the syntax you are parsing? So it is clear for users that you are parsing `(` and `)` as well if present, and that you are returning `[]` if nothing was parsed.

Reply:
I have updated the name and the doc. I'm not sure the parse format is correct or not?